ntirpc/rpc/gss_internal.h

```diff
@@ -113,30 +112,17 @@ svc_rpc_gss_data *alloc_svc_rpc_gss_data(void)
 }
 
 static inline void
-unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd,
-		       uint32_t flags)
+unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd)
 {
-	u_int refcnt;
-	bool gd_locked = flags & SVC_RPC_GSS_FLAG_LOCKED;
-
-	refcnt = atomic_dec_uint32_t(&gd->refcnt);
+	mutex_lock(&gd->lock);
```

Comment:
The decision that this structure has to be lock protected was made before this patch.

Here, the lintention of the ock is not to protect the atomic_dec but to have 
the entire operation (unref and possibly destroy) being "atomic".
If you don't do that, another code path might be able to fetch gd from req via auth and use while we are in the process of destroying.
The observation that this is not required in this very special case with our current code is not relevant. If we have a lock / refcount protected structure we must the lock / refcount always.